geopolars/src/geoseries.rs

```diff
@@ -31,12 +31,28 @@ pub trait GeoSeries {
     /// geometry.
     fn envelope(&self) -> Result<Series>;
 
+    /// Returns a Series with the value of the euclidean length of each geometry
+    ///
+    /// Calculates the euclidean length of each geometry in the series and returns it as a series.
+    /// Not valid for Point or MultiPoint geometries. For Polygon it's the
+    /// length of the exterior ring of the exterior ring of the Polygon and for MultiPolygon
+    /// it returns the
+    fn euclidean_length(&self) -> Result<Series>;
+
     /// Returns a GeoSeries of LinearRings representing the outer boundary of each polygon in the
     /// GeoSeries.
     ///
     /// Applies to GeoSeries containing only Polygons. Returns `None` for other geometry types.
     fn exterior(&self) -> Result<Series>;
 
+    /// Returns a Series with the value of the geodesic length of each geometry
```

Comment:
I don't really know: are geodesic and haversine lengths the same concept, just that geodesic is more accurate? I.e. euclidean length returns values in the same CRS as input, but geodesic and haversine both take as input spherical coordinates and return distances in meters?

Just curious because I'm wondering if maybe the API should be `geodesic_length(method='haversine')` (maybe just in the Python API, assuming that people using the rust api would be familiar with `geo`)

Reply:
Yeah I was tempted to go that way as well. They are as far as I know all just different methods to calculate the same thing.

I figure it doesn't hurt to have them all in there but we should perhaps create a wrapper  method both in rust and python that defaults to whatever one we think is the best to use.

I actually missed one of the implementations: Vincenty’s formula which I can add quickly.

Reply:
Refactored this have a single method that takes a GeodesicLengthMethod argument.